meta-mender-tegra/scripts/mender_tegra_test.py

```diff
@@ -0,0 +1,263 @@
+            result = conn.run(f'df -h | grep {result.stdout.strip()}')
+            if result.return_code != 0:
+                raise RuntimeError("Boot and Rootfs Partition Mismatch detected")
+        else:
+            raise RuntimeError("Cannot Identify Rootfs Partition Slot")
+
+    def verify_slot_change_across_reboots(self):
+        prev_boot_slot = self.nvbootctrl_current_slot()
+        self.reboot()
+        boot_slot = self.nvbootctrl_current_slot()
+        if boot_slot != prev_boot_slot:
+           raise RuntimeError("Boot slot changed from {} to {} across reboots".format(prev_boot_slot,boot_slot))
+
+    def verify_slot_change_across_mender_update(self):
+        prev_boot_slot = self.nvbootctrl_current_slot()
+        self.reboot()
+        boot_slot = self.nvbootctrl_current_slot()
+        if boot_slot == prev_boot_slot:
+           raise RuntimeError("Boot slot not changed from {} to {} after mender update reboot".format(prev_boot_slot,boot_slot))
+
+    def do_single_mender_update(self, rollback_flag):
+        # Connecting to the device
+        self.wait_for_device()
+        # mender install 
+        self.mender_install()
+        # Make sure the sentinel file is not present before reboot for successful boot cases.
+        # sentinel file required for rollback cases
+        if rollback_flag:
+            self.add_sentinel_file()
+        else:
+            self.remove_sentinel_file()
+
+        # Check Partition Mismatch
+        self.check_partition_mismatch()
+        # Check slot changed after a mender update followed by reboot.
+        self.verify_slot_change_across_mender_update()
+
+        # rollback_flag is set true to replicate rollback case, false to replicate successful mender update case.
+        # Mender commit is redundant in case of cboot(no harm in commiting or not), but we need to make sure we don't mender commit when testing rollback case for uboot.
+        # Since that would lead to partition mismatch.
+        # In conclusion for uboot, mender commit is required for a successful mender update and not required to test the rollback case.
+        if not rollback_flag:
+            self.mender_commit()
+
+    def check_rollback(self):
+        boot_slot = self.nvbootctrl_current_slot()
+
+        # nvidia gives 7 retries before switching boot partition
+        for i in range(7):
+            print(f'Starting reboot {i} to test rollback case')
+            # Check Partition Mismatch
+            self.check_partition_mismatch()
+            # check the current boot_slot
+            if boot_slot != self.nvbootctrl_current_slot():
+                self.count += 1 
+            self.reboot()
+        # Device expected to rollback here
+        if boot_slot != self.nvbootctrl_current_slot():
+            print("Success: Rollback after 7 reboots")
+        else:
+            raise RuntimeError("ERROR: No rollback after 7 reboots")
+        self.remove_sentinel_file()
```

Comment:
Not sure why we need to remove_sentinel_file here, I suspect it's because we aren't removing it where we add it.  It would be more clear to remove and add in the same method.

Reply:
We were removing where the file was added, just did not get rid of this line.